internal/services/netapp/netapp_volume_group_sap_hana_resource.go

```diff
@@ -0,0 +1,641 @@
+			volumeClient := metadata.Client.NetApp.VolumeClient
+
+			id, err := volumegroups.ParseVolumeGroupID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			metadata.Logger.Infof("Decoding state for %s", id)
+			var state NetAppVolumeGroupSapHanaModel
+			if err := metadata.Decode(&state); err != nil {
+				return err
+			}
+
+			metadata.Logger.Infof("Updating %s", id)
+
+			if metadata.ResourceData.HasChange("volume") {
+
+				// Iterating over each volume and performing individual patch
+				for i := 0; i < metadata.ResourceData.Get("volume.#").(int); i++ {
+
+					// Checking if individual volume has a change
+					volumeItem := fmt.Sprintf("volume.%v", i)
+
+					capacityPoolId, err := capacitypools.ParseCapacityPoolID(metadata.ResourceData.Get(fmt.Sprintf("%v.capacity_pool_id", volumeItem)).(string))
+					if err != nil {
+						return err
+					}
+
+					if metadata.ResourceData.HasChange(volumeItem) {
+
+						volumeId := volumes.NewVolumeID(id.SubscriptionId,
+							id.ResourceGroupName,
+							id.NetAppAccountName,
+							capacityPoolId.CapacityPoolName,
+							metadata.ResourceData.Get(fmt.Sprintf("%v.name", volumeItem)).(string))
+
+						update := volumes.VolumePatch{
+							Properties: &volumes.VolumePatchProperties{},
+						}
+
+						if metadata.ResourceData.HasChange(fmt.Sprintf("%v.storage_quota_in_gb", volumeItem)) {
+							storageQuotaInBytes := int64(metadata.ResourceData.Get(fmt.Sprintf("%v.storage_quota_in_gb", volumeItem)).(int) * 1073741824)
+							update.Properties.UsageThreshold = utils.Int64(storageQuotaInBytes)
+						}
+
+						if metadata.ResourceData.HasChange(fmt.Sprintf("%v.export_policy_rule", volumeItem)) {
+							exportPolicyRuleRaw := metadata.ResourceData.Get(fmt.Sprintf("%v.export_policy_rule", volumeItem)).([]interface{})
+
+							// Validating export policy rules
+							volumeProtocolRaw := (metadata.ResourceData.Get(fmt.Sprintf("%v.protocols", volumeItem)).([]interface{}))[0]
+							volumeProtocol := volumeProtocolRaw.(string)
+
+							errors := make([]error, 0)
+							for _, ruleRaw := range exportPolicyRuleRaw {
+								if ruleRaw != nil {
+									rule := volumegroups.ExportPolicyRule{}
+
+									v := ruleRaw.(map[string]interface{})
+									rule.Nfsv3 = utils.Bool(v["nfsv3_enabled"].(bool))
+									rule.Nfsv41 = utils.Bool(v["nfsv41_enabled"].(bool))
+
+									errors = append(errors, validate.ValidateNetAppVolumeGroupExportPolicyRuleSAPHanna(rule, volumeProtocol)...)
+								}
+							}
+
+							if len(errors) > 0 {
+								return fmt.Errorf("one or more issues found while performing export policies validations for %s:\n%+v", id, errors)
+							}
+
+							exportPolicyRule := expandNetAppVolumeGroupVolumeExportPolicyRulePatch(exportPolicyRuleRaw)
+							update.Properties.ExportPolicy = exportPolicyRule
+						}
+
+						if metadata.ResourceData.HasChange(fmt.Sprintf("%v.data_protection_snapshot_policy", volumeItem)) {
```

Comment:
Resolved, since this is a single property (snapshot Id), I made a test where it starts without snapshot policy and then it adds it.

![image](https://user-images.githubusercontent.com/17993275/234148530-a852ad6a-b6bf-40bc-8f17-46847f0e1047.png)